packages/flutter_tools/lib/src/macos/cocoapod_utils.dart

```diff
@@ -43,11 +41,11 @@ Future<void> processPodsIfNeeded(
     logger: globals.logger,
   );
 
-  final bool didPodInstall = await globals.cocoaPods.processPods(
+  final bool didPodInstall = await globals.cocoaPods?.processPods(
```

Comment:
When would `cocoaPods` be null?

Reply:
It wouldn't in a real scenario, but it could be unset in a test `context`.

For example `globals.artifacts` is nullable
https://github.com/flutter/flutter/blob/cf7ed18da1b34df95be6eab94e66b365f97ac483/packages/flutter_tools/lib/src/globals_null_migrated.dart#L39-L40
even though in theory it should always be set:
https://github.com/flutter/flutter/blob/7c5857d3b00f2dba36040d2e68cef79ca58071b1/packages/flutter_tools/lib/src/context_runner.dart#L133

and we also have a few of these that are expected to always be set, even by a test:
https://github.com/flutter/flutter/blob/cf7ed18da1b34df95be6eab94e66b365f97ac483/packages/flutter_tools/lib/src/globals_null_migrated.dart#L41-L42

I've had to track down g3 test failures where a test hit something like `globals.artifacts!` and failed because its context didn't set an artifact, because it was unrelated to the test.

I can make this `globals.cocoaPods!` if you prefer and fix up what breaks, if anything.

Reply:
We should eventually make these non-nullable, but we can do that in the future when more stuff is migrated.